### PR TITLE
Cart: make single-select checkbox look like a button (Z#23121616)

### PR DESCRIPTION
**Pull request by wiffbi**
This PR explores the option to make the checkbox for items with a max-oder of 1 to look like a button. I like the idea, maybe something is missing? Add an icon? Or have it read „select one“?

Note: Firefox currently seems to not support the combination of `:has` and `:checked` – so atm only Safari/Chrome show the darkened background. For testing purposes this is okay. If we go with this design, then we might fallback on JS to handle this or ignore and wait for Firefox to catch up (as it is not that substantial).

Bildschirmfoto 2023-05-24 um 15 36 12
Bildschirmfoto 2023-05-24 um 15 36 47

Maybe with an icon? I like the one with an icon in front, but then we might remove the checkbox and use the icon to indicate state. E.g. an icon of a cart, then when checked, have a pointer pointing inside the cart? I fear using a checkmark on top of a cart might be misunderstood as in the item has already been added to the cart.

Bildschirmfoto 2023-05-24 um 15 40 37
Bildschirmfoto 2023-05-24 um 15 39 00

Just an icon:
Bildschirmfoto 2023-05-24 um 15 46 11
Bildschirmfoto 2023-05-24 um 15 46 32

Also not sure about the wording, whether that needs to change, when the checkbox is checked. „Selected“? Hm, again could add more confusion than it helps.

UPDATE: After thinking about it some more, I think keeping the checkbox is helpful in understanding, what this input element is meant to do. I still like the icon and I think it improves communicating the action better than just „Select“. But checkbox and icon directly next to each other can look a bit crowded. So, how about aligning the checkbox left and centering the icon/text? I think combined with the icon-change from above (pointer-down when checked) that will work best. Updated this PR to the following label-layout:

Bildschirmfoto 2023-06-01 um 08 35 45
Bildschirmfoto 2023-06-01 um 08 36 01

### CODE REVIEW

**Review by raphaelm — COMMENTED** [review]
I like it! Only think I'm a little worried about are if the string gets too long in other languages:

image

File: src/pretix/presale/templates/pretixpresale/event/fragment_addon_choice.html [hunk 1]
```diff
@@ -187,7 +187,8 @@ <h5 id="cp-{{ form.pos.pk }}-item-{{ item.pk }}-{{ var.pk }}-legend">{{ var }}</
                                                         name="cp_{{ form.pos.pk }}_variation_{{ item.id }}_{{ var.id }}"
                                                         data-exclusive-prefix="cp_{{ form.pos.pk }}_variation_{{ item.id }}_"
                                                         aria-label="{% blocktrans with item=item.name var=var %}Add {{ item }}, {{ var }} to cart{% endblocktrans %}">
-                                                <i class="fa fa-cart-plus fa-lg" aria-hidden="true"></i>
+                                                <i class="fa fa-shopping-cart" aria-hidden="true"></i>
+                                                {% trans "Select" %}
```
[the remark this answers is not in the document]
Comment (wiffbi) [reply src/pretix/presale/templates/pretixpresale/event/fragment_addon_choice.html]: Good idea to add the context!

**Comment by wiffbi** [comment]
> I like it! Only think I'm a little worried about are if the string gets too long in other languages:
> 
> image

Yeah, that problem do all buttons currently have – although this button might be one of the smaller one. We could add a `overflow:hidden; text-overflow: ellipsis` to get a better optical result, but not ideal. The good thing is, we also have an icon, which should help in understanding, even if text is cropped.

Should I add `overflow:hidden; text-overflow: ellipsis` to the global `.btn`-class or should I just use it for this particular case on `.item-checkbox-label`? Speaking of naming that class, I think I should rename it to just `.checkbox-button` or even `.btn-checkbox` as it needs the other `.btn-X` classes as well. What do you think?

**Comment by raphaelm** [comment]
I think buttons in other places can even have linebreaks? I'm not sure, I think I haven't seen it before.

I'd only change it for this one, not generally for all buttons.

I have no strong opinion on checkbox-button vs btn-checkbox.

**Comment by wiffbi** [comment]
I will merge tomorrow.

**Pull request merged**